### DISCUSSION TITLE
[question] speed_estimate_period

### DISCUSSION THREAD

**Discussion by WellingtonFonseca** [Textualize/rich#2679] [General] [answered]
Guys, I can't understand. speed_estimate_period is the time I expect each step of the process to take to process, correct? if yes, then something is wrong, because I am programming it to wait an hour and it is showing me 20 minutes

Thanks in advance

**Comment by willmcgugan** [comment, marked as answer]
`speed_estimate_period` is the time over which the speed is calculated. It should generally be enough to cover at least 2 samples.

**Reply by WellingtonFonseca** [reply]
they are just for me to understand better, in the case if set to 2, it waits for two steps to be executed, takes the average time of these two and multiplies it by the total steps, is that it?

**Reply by willmcgugan** [reply]
No, its a time, in seconds. samples that occur within this period are used to calculate the speed.

**Reply by WellingtonFonseca** [reply]
ah yes, now i understand. the time for the estimated time recalculation to be performed.

**Comment by WellingtonFonseca** [comment]
is there a possibility to manually enter the expected time? because I have an expectation and if that expectation is exceeded, it means that there is something wrong with the process, you know?
